index.bs

```diff
@@ -1215,6 +1215,383 @@ implementation of this step is specific to the user agent.", the controlling
+text track data.  A series of data frames that share a data type and timeline
+form a data encoding.
+
+To allow for data frames to be sent out of order, they may be sent in separate
+QUIC streams, but more than one data frame may be sent in one QUIC stream if
+that makes sense for a specific type of data.
+
+Text track data uses a payload type of text, a default duration of unknown, and
+a timescale of 1000000 (microseconds).
+
+: encoding-id
+:: Identifies the media encoding to which this data frame belongs.  This can be
+    used to reference properties of the encoding such as the type of data and
+    time scale.  Referencing properties of the encoding through the encoding id
+    helps to avoid sending duplicate information in every frame.
+
+: sequence-number
+:: Identifies the frame and its order in the encoding.
+    Within an encoding, larger sequence numbers mean later start times.
+    Within an encoding, gaps in sequence numbers mean frames are missing.
+
+: start-time
+:: Identifies the beginning of the time range of the data frame.  The time
+    scale is inferred from the properties of the encoding.  The end time can be
+    inferred from the start time and duration.
+
+: duration
+:: If present, the duration of the data frame.  The time
+    scale is inferred from the properties of the encoding.  Likewise, if not
+    present, the duration is inferred from the properties of the encoding.
+
+: sync-time
+:: If present, a time used to synchronize the start time of this audio frame (and
+    thus, this encoding) with that of other media encodings on different
+    timelines.
+
+: payload
+:: The encoded audio samples.  The codec and codec parameters are inferred from
+    the properties of the encoding.
+
+Feedback {#streaming-feedback}
+------------------------------------
+
+The receiver can send feedback to the sender, such as key frame requests.
+
+A video key frame is requested by sending a video-request message with
+the following keys and values.
+
+To allow for video frames to be sent out of order, they may be sent in separate
+QUIC streams.
+
+: encoding-id
+:: The encoding for which the sender should send a new key frame.
+
+: sequence-number
+:: Gives the order in the encoding.
+    Within an encoding, larger sequence numbers invalidate previous ones.
+    A sender may ignore smaller sequence numbers after a larger one has been processed.
+    This it to prevent out-of-order requests from generating more key frames than necessary. 
+
+: highest-decoded-frame-sequence-number: uint
+:: If set, the sender may generate a video frame dependent on the last decoded
+    frame.  If not set, the sender must generate an indepdendent (key) frame.
+
+Stats {#streaming-stats}
+------------------------------
+
+<!-- 
+
+ TODO: Something like this:
+
+streaming-session-sender-stats-event = {
+  timestamp: microseconds
+  ? audio: [* {
+    encoding-id: uint
+    ? sent: microseconds
+    ? encode-delay: microseconds
```

Comment:
Here and below: The spec (and generated code) would be clearer if this were pulled out into its own type

Reply:
Done